python-checks/src/main/java/org/sonar/python/checks/JwtVerificationCheck.java

```diff
@@ -175,4 +186,83 @@ private static boolean isCallToVerifyJwt(Tree t) {
       .isPresent();
   }
 
+  private static boolean accessOnlyAllowedHeaderKeys(CallExpression call) {
+    Tree assignment = TreeUtils.firstAncestorOfKind(call, Tree.Kind.ASSIGNMENT_STMT);
+    if (assignment != null) {
+      List<Expression> lhsExpressions = ((AssignmentStatement) assignment).lhsExpressions().stream()
+        .map(ExpressionList::expressions)
+        .flatMap(Collection::stream).toList();
+      if (lhsExpressions.size() == 1 && lhsExpressions.get(0).is(Tree.Kind.NAME)) {
+        Name name = (Name) lhsExpressions.get(0);
+        var symbol = name.symbol();
+        if (symbol != null) {
+          var argumentsOfGet = argumentsAccessedByGet(symbol, call);
+          var argumentsOfSubscription = argumentsAccessedBySubscription(symbol, call);
+          return areStringLiteralsPartOfAllowedKeys(Stream.concat(argumentsOfGet, argumentsOfSubscription));
+        }
+      }
+    }
+    return false;
+  }
+
+  private static boolean areStringLiteralsPartOfAllowedKeys(Stream<StringLiteral> literals) {
+    var literalList = literals.toList();
+    return !literalList.isEmpty() && literalList.stream().allMatch(str -> ALLOWED_KEYS_ACCESS.contains(str.trimmedQuotesValue()));
+  }
+
+  private static Stream<StringLiteral> argumentsAccessedByGet(Symbol symbol, CallExpression call) {
+    var usages = getHeaderDictUsages(symbol, call);
+    var parentOfUsages = usages.map(Usage::tree).map(Tree::parent);
+    var callExpressionsFromUsages = getCallExprWhereDictIsAccessedWithGet(parentOfUsages);
+    var callExpressionFromGetUnverifiedHeaders = getCallExprWhereDictIsAccessedWithGet(Stream.of(call.parent()));
+    var arguments = Stream.concat(callExpressionsFromUsages, callExpressionFromGetUnverifiedHeaders).map(CallExpression::arguments).flatMap(Collection::stream);
+    return getStringLiteralArguments(arguments);
+  }
+
+  private static Stream<Usage> getHeaderDictUsages(Symbol symbol, CallExpression call) {
+    return symbol.usages().stream()
+      .filter(usage -> usage.tree().firstToken().line() > call.callee().firstToken().line());
+  }
+
+  private static Stream<CallExpression> getCallExprWhereDictIsAccessedWithGet(Stream<Tree> parentQualifiedExpr) {
+    return parentQualifiedExpr
+      .filter(parent -> parent.is(Tree.Kind.QUALIFIED_EXPR))
+      .map(TreeUtils.toInstanceOfMapper(QualifiedExpression.class))
```

Comment:
Can be replaced with the following:
```suggestion
      .flatMap(TreeUtils.toStreamInstanceOfMapper(QualifiedExpression.class))
```